packages/credential-providers/src/fromAwsCliV2CompatibleProviderChain.ts

```diff
@@ -0,0 +1,134 @@
+import { CredentialsProviderError } from "@smithy/property-provider";
+import type { AwsCredentialIdentity, Logger } from "@smithy/types";
+
+/**
+ * @public
+ */
+
+export type AwsCliV2CompatibleProviderOptions = Partial<AwsCredentialIdentity> &
+  FromEnvInit &
+  FromIniInit &
+  RemoteProviderInit &
+  FromProcessInit &
+  FromSSOInit &
+  FromTokenFileInit & {
+    /**
+     * Setting a client profile is similar to setting a value for the
+     * AWS_PROFILE environment variable. Setting a profile on a client
+     * in code only affects the single client instance, unlike AWS_PROFILE.
+     *
+     * When set, and only for environments where an AWS configuration
+     * file exists, fields configurable by this file will be retrieved
+     * from the specified profile within that file.
+     * Conflicting code configuration and environment variables will
+     * still have higher priority.
+     *
+     * For client credential resolution that involves checking the AWS
+     * configuration file, the client's profile (this value) will be
+     * used unless a different profile is set in the credential
+     * provider options.
+     *
+     */
+    profile?: string;
+
+    /**
+     * Optional logger for logging debug/info/warn/error.
+     */
+    logger?: Logger;
+  };
+
+/**
+ * @public
+ *
+ * Creates a credential provider that sources credentials using the same priority
+ * chain as the AWS CLI v2:
+ *
+ * 1. Static credentials from initialization
+ * 2. Profile credentials (if profile specified)
+ * 3. Environment variables
+ * 4. Web Identity Token credentials
+ * 5. SSO credentials
+ * 6. Process credentials
+ * 7. Remote credentials (ECS, EC2 Instance Metadata)
+ *
+ * Uses dynamic imports and `createCredentialChain` to mimic AWS CLI V2 behavior.
+ *
+ * @param init - Configuration options for the provider chain
+ * @returns An AWS credential provider function that returns a promise for credentials
+ */
+
+export const fromAwsCliV2CompatibleProviderChain =
+  (_init: AwsCliV2CompatibleProviderOptions = {}): RuntimeConfigAwsCredentialIdentityProvider =>
+  async ({ callerClientConfig } = {}): Promise<AwsCredentialIdentity> => {
+    // Merge init with caller's client config (profile/region).
+    const init: AwsCliV2CompatibleProviderOptions = {
+      ..._init,
+      ...callerClientConfig,
+      logger: _init.logger ?? callerClientConfig?.logger,
+    };
+
+    init.logger?.debug(
+      "@aws-sdk/credential-providers - fromAwsCliV2CompatibleProviderChain - Initializing credential chain"
+    );
+
+    const { profile, logger, ...awsCredentials } = init;
+
+    // 1. If credentials are explicitly provided, return them.
+    if (awsCredentials.accessKeyId && awsCredentials.secretAccessKey) {
+      logger?.debug(
+        "@aws-sdk/credential-providers - fromAwsCliV2CompatibleProviderChain - using static credentials from initialization"
+      );
+      return awsCredentials as AwsCredentialIdentity;
```

Comment:
extract only the fields related to `AwsCredentialIdentity` from `...awsCredentials`, since it contains all possible fields from the intersection type.